test/example.c

```diff
@@ -19,13 +19,94 @@
 #  define TESTFILE "foo.gz"
 #endif
 
-#define CHECK_ERR(err, msg) { \
-    if (err != Z_OK) { \
-        fprintf(stderr, "%s error: %d\n", msg, err); \
-        exit(1); \
+#define SUCCESSFUL                 1
+#define FAILED_WITH_ERROR_CODE     0
+#define FAILED_WITHOUT_ERROR_CODE -1
+
+typedef struct test_result_s {
+    int           result; /* One of: SUCCESSFUL,
+                          FAILED_WITH_ERROR_CODE, or
+                          FAILED_WITHOUT_ERROR_CODE*/
+    int           error_code; /* error code if success is FAILED_WITH_ERROR_CODE */
+    int           line_number;
+    z_const char* message;
+    z_const char* extended_message;
```

Comment:
gzerror's return type is const char *, not z_const char *, leading to warnings?

Reply:
I think I misunderstood the purpose of z_const. I think you're right. I'll make that change.